### PR TITLE
feat(timer): Timer component

### PR DESCRIPTION
**Pull request by ChefHutch**
This is a generic timer component that will render a D, H, M countdown and a tooltip.

UI Example with tooltip while hovered
Screenshot 2021-04-01 at 15 15 12

UI example without tooltip
Screenshot 2021-04-01 at 15 15 32

UI example without using custom text components
Screenshot 2021-04-01 at 15 16 45

Code usage:
```js
  <Timer
    timerStage={'string that preceeds the timer'}
    minutes={'minutes value as number'}
    hours={'hours value as number'}
    days={'days value as number'}
    blockNumber={'number used by the tooltip to show the specific block being counted to'}
    showTooltip={'boolean - render or don't render tooltip'}
    HeadingTextComponent={'component - allows you to pass a custom header text comp, with sensible defaults'}
    BodyTextComponent={'component - allows you to pass a custom body text comp, with sensible defaults'}
  />
```

Code usage example:

```js
const TimerHeadingComponent = ({ children }) => (
  <StyledHeading background="linear-gradient(180deg, #FFD800 0%, #EB8C00 100%)" fill>
    {children}
  </StyledHeading>
)

const TimerBodyComponent = ({ children }) => (
  <Text color="#ffff" fontWeight="600" fontSize="16px" mr="16px">
    {children}
  </Text>
)

const App = ({minutes, hours, days, targetBlockNumber}) => {
...
...
...
  return (
    <Timer
      timerStage={
        competitionState.state === LIVE ? `${TranslateString(410, 'End')}:` : `${TranslateString(1212, 'Start')}:`
      }
      minutes={minutes}
      hours={hours}
      days={days}
      blockNumber={targetBlockNumber}
      showTooltip
      HeadingTextComponent={TimerHeadingComponent}
      BodyTextComponent={TimerBodyComponent}
    />
  )
}
```

### CODE REVIEW

File: src/views/Farms/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -44,11 +45,11 @@ const Container = styled.div`
   }
 `
 
-const Tooltip: React.FunctionComponent<TooltipProps> = ({ content, children }) => {
+const Tooltip: React.FunctionComponent<TooltipProps> = ({ content, children, className }) => {
   return (
     <Container>
       {children}
-      <TooltipContent>{content}</TooltipContent>
+      <TooltipContent className={className}>{content}</TooltipContent>
```
Comment (ChefHutch) [inline src/views/Farms/components/Tooltip/Tooltip.tsx:52]: Required to allow tooltip to be restyled here https://github.com/pancakeswap/pancake-frontend/pull/773/files#diff-7139049a58c241ef6622e4a8f989946a026906be596a4b442f86a423d493a436R6.

By default it sits above the component, hence the restyling. 

I think work is already being done on this tooltip - hence not going further and making accept a 'position' prop and dynamically style it that way or similar

File: src/components/Timer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+export interface TimerProps {
+  timerStage?: string
+  minutes?: number
+  hours?: number
+  days?: number
+  showTooltip?: boolean
+  blockNumber?: number
+  HeadingTextComponent?: any
+  BodyTextComponent?: any
```
Comment (hachiojidev) [inline src/components/Timer/index.tsx:14]: `ReactElement` is what I think you need here...

File: src/components/Timer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    {children}
+  </Text>
+)
+
+const Wrapper: React.FC<TimerProps> = ({
+  timerStage,
+  minutes,
+  hours,
+  days,
+  showTooltip,
```
Comment (hachiojidev) [inline src/components/Timer/index.tsx:53]: This should have a default value.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

**Review by RabbitDoge — COMMENTED** [review]
The tooltip system is something we will reuse and should be in the uikit. So far there is only one direction, but maybe we can make a good use of popperjs later. Anyway, if you can start by making this one generic and add it to the uikit it would be a good start.

File: src/components/Timer/index.tsx (hunk 2 above)
Comment (ChefHutch) [reply src/components/Timer/index.tsx:14]: Dead right. I left this in by accident - was experimenting with ways of passing a component as props it.

**Comment by ChefHutch** [comment]
Made those changes @hachiojidev  

As discussed with @RabbitDoge - we'll look at moving this into the uikit after the trading competition (priority) is live.

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**